craft_providers/base.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2021 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Base configuration module."""
+import logging
+from abc import ABC, abstractmethod
+from typing import Optional
+
+from .executor import Executor
+
+logger = logging.getLogger(__name__)
+
+
+class Base(ABC):
+    """Base definition for bootstrap configuration and setup.
+
+    :param name: Name of base.
+    """
+
+    def __init__(self, *, name: str) -> None:
+        self.name = name
+
+    @abstractmethod
+    def setup(self, *, executor: Executor) -> None:
+        """Configure instance as necessary.
```

Comment:
```suggestion
        """Configure instance with the necessary environment.
```

But it begs the question, what is the necessary environment, or what should classes inheriting from this one implement